src/commands/app/db/collection/updateMany.js

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2025 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { DBBaseCommand } from '../../../../DBBaseCommand.js'
+import { Args, Flags } from '@oclif/core'
+import chalk from 'chalk'
+import { asObject, isNonEmptyString } from '../../../../utils/inputValidation.js'
+
+export class UpdateMany extends DBBaseCommand {
+  async run () {
+    const { collectionName, filter, update } = this.args
+    const { upsert } = this.flags
+
+    try {
+      this.log(chalk.blue(`Updating documents in collection '${collectionName}'...`))
+
+      // Log filter and update if provided
+      this.log(chalk.dim(`   Filter: ${JSON.stringify(filter)}`))
+      this.log(chalk.dim(`   Update: ${JSON.stringify(update)}`))
+
+      if (upsert) {
+        this.log(chalk.dim('   Upsert enabled: Will create document if not found'))
+      }
+
+      const client = await this.db.connect()
+      const collection = await client.collection(collectionName)
+
+      // Build options object
+      const updateOptions = {}
+      if (upsert) {
+        updateOptions.upsert = true
+      }
+
+      // Perform the updateMany operation
+      const result = await collection.updateMany(filter, update, updateOptions)
+
+      this.debugLogger?.info?.('Documents updated successfully:', result)
+
+      const response = {
+        collectionName,
+        matchedCount: result.matchedCount,
+        modifiedCount: result.modifiedCount,
+        upsertedCount: result.upsertedCount,
+        upsertedId: result.upsertedId,
+        status: 'updated',
+        namespace: this.rtNamespace,
+        timestamp: new Date().toISOString(),
+        result
+      }
+
+      if (upsert) {
+        response.upsert = true
+        response.options = updateOptions
+      }
+
+      this.log(chalk.green(`Successfully updated documents in collection '${collectionName}'`))
+      this.log(chalk.dim(`   Collection: ${collectionName}`))
+      this.log(chalk.dim(`   Matched Count: ${result.matchedCount}`))
+      this.log(chalk.dim(`   Modified Count: ${result.modifiedCount}`))
+      this.log(chalk.dim(`   Namespace: ${this.rtNamespace}`))
+
+      if (result.upsertedCount > 0) {
+        this.log(chalk.dim(`   Upserted Count: ${result.upsertedCount}`))
+        if (result.upsertedId) {
+          this.log(chalk.dim(`   Upserted ID: ${result.upsertedId}`))
+        }
+      }
```

Comment:
An upsert is only performed if no documents matching the filter were found.

```suggestion
      if (result.upsertedCount) {
        this.log(chalk.green(`No documents in collection '${collectionName}' were found matching the filter, performed an upsert instead`))
        this.log(chalk.dim(`   Collection: ${collectionName}`))
        this.log(chalk.dim(`   Upserted ID: ${result.upsertedId}`))
        this.log(chalk.dim(`   Namespace: ${this.rtNamespace}`))
      } else {
        this.log(chalk.green(`Successfully updated documents in collection '${collectionName}'`))
        this.log(chalk.dim(`   Collection: ${collectionName}`))
        this.log(chalk.dim(`   Matched Count: ${result.matchedCount}`))
        this.log(chalk.dim(`   Modified Count: ${result.modifiedCount}`))
        this.log(chalk.dim(`   Namespace: ${this.rtNamespace}`))
      }
```